src/components/sprint-view-table/tasks-table.tsx

```diff
@@ -46,45 +47,57 @@ const TaskTable = ({project, loggedInpersonId, filter}: Props) => {
   /**
    * Handle loggenInPersonId change
    */
-  useEffect(()=>{
+  useEffect(() => {
     setReload(true);
-  },[loggedInpersonId]);
-  
+  }, [loggedInPersonId]);
+
   /**
-   * Get tasks and total time entries  
+   * Get tasks and total time entries
    */
   const getTasksAndTimeEntries = async () => {
     setLoading(true);
-    if (reload || !tasks.length || !timeEntries.length){
+    if (reload || !tasks.length || !timeEntries.length) {
       try {
-        let fetchedTasks = await tasksApi.listProjectTasks({projectId: project.id});
-        if (loggedInpersonId) {
-          fetchedTasks = fetchedTasks.filter((task)=> task.assignedPersons?.includes(loggedInpersonId ));
-        }
+        const fetchedTasks = !loggedInPersonId ? await tasksApi.listProjectTasks({ projectId: project.id })
+        : await tasksApi.listProjectTasks({ projectId: project.id }).then(result => result.filter((task) =>
+          task.assignedPersons?.includes(loggedInPersonId || 0)
+        ));
+
         setTasks(fetchedTasks);
-        const fetchedTimeEntries = await Promise.all(fetchedTasks.map(async (task) => {
-          try {
-            if (project.id ){           
-              const totalTimeEntries = await timeEntriesApi.listProjectTimeEntries({ projectId: project.id, taskId: task.id });
-              let totalMinutes = 0;
-              totalTimeEntries.forEach((timeEntry: TimeEntries)=> {
-                if (loggedInpersonId && timeEntry.person===loggedInpersonId) {
-                  totalMinutes+=(timeEntry.timeRegistered || 0)
-                }
-                if (!loggedInpersonId) {
-                  totalMinutes += timeEntry.timeRegistered || 0;
-                }
-              })
-              return totalMinutes;
+        const fetchedTimeEntries = await Promise.all(
+          fetchedTasks.map(async (task) => {
+            try {
+              if (project.id) {
+                const totalTimeEntries = await timeEntriesApi.listProjectTimeEntries({
+                  projectId: project.id,
+                  taskId: task.id
+                });
+                let totalMinutes = 0;
+                totalTimeEntries.forEach((timeEntry: TimeEntries) => {
+                  if (loggedInPersonId && timeEntry.person === loggedInPersonId) {
+                    totalMinutes += timeEntry.timeRegistered || 0;
+                  }
+                  if (!loggedInPersonId) {
+                    totalMinutes += timeEntry.timeRegistered || 0;
+                  }
+                });
+                return totalMinutes;
+              }
+            } catch (error) {
+              if (task.id) {
```

Comment:
We still want an error message even if there is no task Id, so perhaps the condtional should be removed but the contents stay the same, Rather than task.id || 0 could something other than 0 be passed e.g. a localized "task id not found"